Source/JavaScriptCore/wasm/WasmBBQPlan.cpp

```diff
@@ -265,8 +265,22 @@ void BBQPlan::didCompleteCompilation()
 
             computePCToCodeOriginMap(context, linkBuffer);
 
+            if (shouldDumpDisassemblyFor(CompilationMode::BBQMode)) {
+                auto* disassembler = context.procedure->code().disassembler();
+
+                const char* airPrefix = "Air" "    ";
+                const char* asmPrefix  = "asm" "          ";
+
+                auto forEachInst = scopedLambda<void(B3::Air::Inst&)>([&] (B3::Air::Inst& inst) {
+                    if (inst.origin)
+                        dataLogLn(*inst.origin);
+                });
+
+                disassembler->dump(context.procedure->code(), WTF::dataFile(), linkBuffer, airPrefix, asmPrefix, forEachInst);
+            }
+
             function->entrypoint.compilation = makeUnique<Compilation>(
-                FINALIZE_CODE(linkBuffer, JITCompilationPtrTag, "WebAssembly BBQ function[%i] %s name %s", functionIndex, signature.toString().ascii().data(), makeString(IndexOrName(functionIndexSpace, m_moduleInformation->nameSection->get(functionIndexSpace))).ascii().data()),
+                FINALIZE_CODE_IF(false, linkBuffer, JITCompilationPtrTag, "WebAssembly BBQ function[%i] %s name %s", functionIndex, signature.toString().ascii().data(), makeString(IndexOrName(functionIndexSpace, m_moduleInformation->nameSection->get(functionIndexSpace))).ascii().data()),
```

Comment:
Is there a better way?

Reply:
I think it is OK to just use the `FINALIZE_CODE`, as what we are doing in FTL is the same.